### PR TITLE
:tada: feat(checkpoint): migrate format :tada: 

### PR DESCRIPTION
**Pull request by PierrickV**
Diff:
- improved parsing of Generic events & Syslog events (grok replace by KV and DSV for more reliable parsing)
- no call to `lower`

This work aims at being ISO with the current parser (no improvements, ECS categorization fields, etc.).

Changelog


| Before update                                                                                                           | After Update                                                                                                                                                                                                                                                                     |
| ----------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `source.port` is a float                                                                                                | `source.port` is an int                                                                                                                                                                                                                                                          |
| `source.nat.port` is a float                                                                                            | `source.nat.port` is an int                                                                                                                                                                                                                                                      |
| `destination.port` is a float                                                                                           | `destination.port` is an int                                                                                                                                                                                                                                                     |
| `destination.port` is a float                                                                                           | `destination.port` is an int                                                                                                                                                                                                                                                     |
| `action.properties.originsicname: 'CN\\=ertfw01,O=foomgmt.foobar.local.zazgch'`                                         | `'originsicname': 'CN=ertfw01,O=foomgmt.foobar.local.zazgch'`                                                                                                                                                                                                                    |
| `action.properties` is of type list[dict]                                                                               | `action.properties` is of type dict (no impact expected)                                                                                                                                                                                                                         |
| `network.transport` was not always translated to the equivalent string value (ex. 1=>imcp)                              | `network.transport` is always translated to string                                                                                                                                                                                                                               |
| truncated content for `action.properties.encryption_methods`, `action.properties.ike_ids`, `action.properties.ike_mode` | the content of the field is complete (e.g. `ESP` become `ESP: AES-256 + SHA1 + PFS (group 5)`                                                                                                                                                                                    |
| fix user_agent parsing of logs type `geo_protection` (e.g. previously `user_agent.original: Google Chrome`)             | `user_agent.device.name: Other user_agent.name: Chrome user_agent.original: Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36 user_agent.os.name: Windows user_agent.os.version: 10 user_agent.version: 88.0.4324 ` |

### CODE REVIEW

File: Checkpoint/checkpoint/tests/Generic_tcp_allow.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "event": {
+      "created": "2019-12-10T23:44:49.000Z",
+      "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430"
+    },
+    "log": {
+      "hostname": "FOOBAR-HOST-01"
+    }
+  },
+  "expected": {
+    "message": "\"22\" \"10Dec2019\" \"23:44:49\" \"eth1\" \"FOOBAR-HOST-01\" \"Log\" \"Accept\" \"443\" \"44250\" \"2.2.2.2\" \"1.1.1.1\" \"tcp\" \"81\" \"WebAccess Notes\" \"81-FooBar_Policy\" \"\" \"inzone: External; outzone: Internal; service_id: https\" \"Multi-product\" \"\" \"\"",
```
Comment (TOUFIKIzakarya) [inline Checkpoint/checkpoint/tests/Generic_tcp_allow.json:13]: Can we parse `timestamp` ?

File: Checkpoint/checkpoint/tests/Generic_tcp_drop.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "event": {
+      "created": "2021-04-23T20:02:05.017771Z",
+      "id": "10f0afe9-98a1-4226-a6bd-8f70d461d430"
+    },
+    "log": {
+      "hostname": "FOOBAR-HOST-01"
+    }
+  },
+  "expected": {
+    "message": "\"21\" \"10Dec2019\" \"23:44:49\" \"eth1\" \"FOOBAR-HOST-01\" \"Log\" \"Drop\" \"58339\" \"46193\" \"1.1.1.1\" \"2.2.2.2\" \"tcp\" \"107\" \"Deny All\" \"107-FooBar_Policy\" \"\" \"inzone: External; outzone: Internal\" \"Security Gateway/Management\" \"\" \"\"",
```
Comment (TOUFIKIzakarya) [inline Checkpoint/checkpoint/tests/Generic_tcp_drop.json:13]: Can we parse `timestamp` ?

**Review by TOUFIKIzakarya — COMMENTED** [review]
Thanks for the job. Little remarks.

**Comment by PierrickV** [comment]
@TOUFIKIzakarya: Thank you for taking the time to review my work :pray: 

We do not parse the `timestamp` because the previous parser did not. The migration does not aim at improving the parser it is just about changing the underlining technology. We want to be able to deploy the exact same behavior to production. The improvements will be carried by the @SEKOIA-IO/sekoia-io-integration-team once the migration is finished.

If you see that the `timestamp` was present in the previous logs it is due to an unrelated behavior related to Syslog (same applies for `host.name` & `host.hostname`).

Have a great day :sun_with_face:

**Review by TOUFIKIzakarya — APPROVED** [review]
LGFM

File: Checkpoint/checkpoint/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+          action.properties.vpn_feature_name: "{{checkpoint.message.vpn_feature_name}}"
+          action.properties.vpn_peer_gateway: "{{checkpoint.message.peer_gateway}}"
+          destination.address: "{{checkpoint.message.dst}}"
+          destination.domain: "{{checkpoint.message.dhost}}"
+          destination.ip: "{{checkpoint.message.dst}}"
+          destination.nat.ip: "{{checkpoint.message.destinationTranslatedAddress or checkpoint.message.xlatedst}}"
+          destination.nat.port: "{{checkpoint.message.destinationTranslatedPort or checkpoint.message.xlatedport}}"
+          destination.port: "{{checkpoint.message.service or checkpoint.message.dpt}}"
+          destination.user.name: "{{checkpoint.message.duser}}"
+          event.code: "{{checkpoint.message.DeviceVendor or checkpoint.message.loguid or checkpoint.message.DeviceEventClassID}}"
```
Comment (squioc) [inline Checkpoint/checkpoint/ingest/parser.yml:119]: I suggest using `DeviceEventClassId` first and remove `DeviceVendor`
```suggestion
          event.code: "{{checkpoint.message.DeviceEventClassID or checkpoint.message.loguid}}"
```

File: Checkpoint/checkpoint/_meta/fields.yml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+source.size_in_char:
+  description: ""
+  name: source.size_in_char
+  type: number
+
+"{{checkpoint.message.proto | lower}}":
+  description: ""
+  name: "{{checkpoint.message.proto | lower}}"
+  type: keyword
```
[the remark this answers is not in the document]
Comment (PierrickV) [reply Checkpoint/checkpoint/_meta/fields.yml]: This should be fixed :pray:

[the remark this answers is not in the document]
Comment (PierrickV) [reply Checkpoint/checkpoint/_meta/fields.yml]: associated commit https://github.com/SEKOIA-IO/intake-formats/pull/800/commits/e7cf716bb64b8fb487da96ca33ca00346ad3baae

**Review by squioc — APPROVED** [review]
LGTM

**Review by squioc — APPROVED** [review]

**Pull request merged**